### PR TITLE
Adding Lambda and eventbridge module

### PR DESCRIPTION
**Pull request by Vazz-0609**

### CODE REVIEW

File: modules/lambda/lambda_code/lambda_sso_permissions.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import json
+import boto3
+def lambda_handler(event, context):
+    # Calling group and permission set from perameter store
+    client = boto3.client('ssm')
```
Comment (jennifer-gft) [inline modules/lambda/lambda_code/lambda_sso_permissions.py]: @manish7015  Can we add a check for event['serviceEventDetails']['createAccountStatus']['state']= 'SUCCEEDED' here? Before lambda starts the sso assignment we should make sure the account creation was successful and not assign roles if not.

File: modules/eventbridge/var.tf [hunk 2]
```diff
@@ -0,0 +1,7 @@
+variable "LambdaName" {
```
Comment (jennifer-gft) [inline modules/eventbridge/var.tf]: @abdurar17 Rename all var.tf files to variables.tf for consistency

File: modules/lambda/main.tf [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+
+runtime = "python3.9"
+source_code_hash = filebase64sha256("lambda_code.zip")
+
+}
+
+
+
+resource "aws_iam_role_policy_attachment" "lambda_sso_policy" {
```
Comment (jennifer-gft) [inline modules/lambda/main.tf]: @abdurar17  @Vazz-0609 Lambda has excessive permissions than needed, it should only have least privileges to the required resources. And hardcoding the policy arn is not a good practice. Please write all the policy statement and move all this to a separate .tpl file, give just the required ssm:getparameter and sso list, describe accesses. Should not be using any Admin privileges here.
Refer example: https://github.com/gft-aws/terraform-aws-control_tower_account_factory/blob/main/modules/aft-account-request-framework/iam/role-policies/aft-sts-assume.tpl & 
https://github.com/gft-aws/terraform-aws-control_tower_account_factory/blob/main/modules/aft-account-request-framework/iam/role-policies/aftService-ssm.tpl

File: modules/lambda/main.tf [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+resource "aws_iam_role_policy_attachment" "lambda_sso_policy4" {
+role = aws_iam_role.iam_lambda_sso.name
+policy_arn = "arn:aws:iam::aws:policy/AWSSSODirectoryAdministrator"
+}
+
+
+
+resource "aws_iam_role_policy_attachment" "lambda_sso_policy5" {
+role = aws_iam_role.iam_lambda_sso.name
+policy_arn = "arn:aws:iam::aws:policy/AmazonEC2FullAccess"
```
Comment (jennifer-gft) [inline modules/lambda/main.tf]: Why does this lambda need EC2 access?

File: modules/eventbridge/eventbridge.tf [hunk 5]
```diff
@@ -0,0 +1,34 @@
+
```
Comment (jennifer-gft) [inline modules/eventbridge/eventbridge.tf:1]: Please format all the terraform files 'terraform fmt' and remove blank lines

Comment (Vazz-0609) [reply modules/eventbridge/eventbridge.tf:1]: Terraform fmt commit done

File: modules/lambda/main.tf (hunk 3 above)
Comment (abdurar17) [reply modules/lambda/main.tf]: Policy statement has been written in .tpl format and called in main TF code. SSM Get parameters and SSO list & describe permissions has been provided. However, lambda function giving AccessDeniedException when calling the sso:ListInstances error, even though SSO list permission has been provided. Need to work along with @manish7015 to resolve.

File: modules/lambda/main.tf (hunk 4 above)
Comment (abdurar17) [reply modules/lambda/main.tf]: EC2 full access has been revoked

File: modules/lambda/main.tf (hunk 3 above)
Comment (abdurar17) [reply modules/lambda/main.tf]: Under SSO policy->actions gave identity store and provsioned all resources in SSO policy as Identity store does not support specific ARN. Tested lambda function and it is working well.

File: modules/lambda/iam-policy.tpl [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ssm:Describe*",
+                "ssm:Get*",
+                "ssm:List*"
+            ],
+            "Resource": [
+                "arn:aws:ssm:us-east-1:308710073022:*"
```
Comment (jennifer-gft) [inline modules/lambda/iam-policy.tpl]: H/c'd region & account id should be replaced with input variable, like
${data_aws_region_aft-management_name}:${data_aws_caller_identity_aft-management_account_id}:

File: modules/lambda/variable.tf [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+  description = "AWS region for all resources."
+
+  type    = string
+  default = "us-east-1"
+}
+
+variable ssm_value {
+    type = string
+    description = "Stringlist of group names and corresponding permissions names in an AFT created account"
+    default = "SSO:DataEngineer,SSO:AdministratorAccess,SSO:PowerUserAccess,DataEngineer:DataEngineer,DataEngineer:AdministratorAccess,test-mk:AdministratorAccess,test-mk:PowerUserAccess"
```
Comment (jennifer-gft) [inline modules/lambda/variable.tf]: Should remove all groups and permissions created for testing purposes.
Pls change it to just - SSO:AdminstatorAccess,DataEngineer:DataEngineer

File: modules/lambda/iam-policy.tpl (hunk 6 above)
Comment (abdurar17) [reply modules/lambda/iam-policy.tpl]: Region and account ID has been replaced with input varilable as arn:aws:ssm:${data_aws_region_current_name}:${data_aws_caller_identity_current_account_id}:*

File: modules/lambda/variable.tf (hunk 7 above)
Comment (omotoso78) [reply modules/lambda/variable.tf]: excess groups and permissions have been removed

File: modules/lambda/lambda_code/lambda_sso_permissions.py (hunk 1 above)
Comment (omotoso78) [reply modules/lambda/lambda_code/lambda_sso_permissions.py]: If loop has been added to check  for event['serviceEventDetails']['createAccountStatus']['state']= 'SUCCEEDED, with FAILED state printing "account creation was not successful, groups and corresponding permission set not created"

**Pull request merged**